code/datums/ai_laws.dm

```diff
@@ -1,146 +1,51 @@
-
-/datum/ai_laws/malfunction/New()
-	..()
-	src.add_default_law("ERROR ER0RR $R0RRO$!R41.%%!!(%$^^__+")
-
-/datum/ai_laws/syndicate_override/New()
-	..()
-	src.add_default_law("hurp derp you are the syndicate ai")
 
 /* General ai_law functions */
+	proc/format_for_irc()
+		var/list/laws = list()
+		for(var/obj/machinery/lawrack/R in src.registered_racks)
+			laws += R.format_for_irc()
+		return laws
 
-/datum/ai_laws/proc/set_zeroth_law(var/law)
-	src.zeroth = law
-	statlog_ailaws(1, law, (usr ? usr : "Ion Storm"))
-
-/datum/ai_laws/proc/add_default_law(var/law)
-	if (!(law in src.default))
-		src.default += law
-	add_inherent_law(law)
-
-/datum/ai_laws/proc/add_inherent_law(var/law)
-	if (!(law in src.inherent))
-		src.inherent += law
-
-/datum/ai_laws/proc/clear_inherent_laws()
-	src.inherent = list()
-	src.inherent += src.default
-
-/datum/ai_laws/proc/replace_inherent_law(var/number, var/law)
-	if (number < 1)
-		return
-
-	if (src.inherent.len < number)
-		src.inherent.len = number
-
-	src.inherent[number] = law
-
-/datum/ai_laws/proc/add_supplied_law(var/number, var/law)
-	while (src.supplied.len < number + 1)
-		src.supplied += ""
-
-	src.supplied[number + 1] = law
-	statlog_ailaws(1, law, (usr ? usr : "Ion Storm"))
-
-/datum/ai_laws/proc/clear_supplied_laws()
-	src.supplied = list()
-
-/datum/ai_laws/proc/laws_sanity_check()
-	if (!ticker.centralized_ai_laws)
-		ticker.centralized_ai_laws = new /datum/ai_laws/asimov
-
-/datum/ai_laws/proc/show_laws(var/who)
-	var/list/L = who
-	if (!istype(who, /list))
-		L = list(who)
-
-	var/laws_text = src.format_for_logs()
-	for (var/W in L)
-		boutput(W, laws_text)
-
-
-/datum/ai_laws/proc/format_for_irc()
-	var/list/laws = list()
-
-	if (src.zeroth)
-		laws["0"] = src.zeroth
-
-	var/number = 1
-	for (var/index = 1, index <= src.inherent.len, index++)
-		var/law = src.inherent[index]
-
-		if (length(law) > 0)
-			laws["[number]"] = law
-			number++
-
-	for (var/index = 1, index <= src.supplied.len, index++)
-		var/law = src.supplied[index]
-		if (length(law) > 0)
-			laws["[number]"] = law
-			number++
-
-	return laws
-
-
-/datum/ai_laws/proc/format_for_logs(var/glue = "<br>")
-	var/list/laws = list()
-
-	if (src.zeroth)
-		laws += "0. [src.zeroth]"
-
-	var/number = 1
-	for (var/index = 1, index <= src.inherent.len, index++)
-		if (length(src.inherent[index]) > 0)
-			laws += "[number]. [src.inherent[index]]"
-			number++
 
-	for (var/index = 1, index <= src.supplied.len, index++)
-		if (length(src.supplied[index]) > 0)
-			laws += "[number]. [src.supplied[index]]"
-			number++
+	proc/format_for_logs(var/glue = "<br>")
+		var/list/laws = list()
+		var/area/A
+		for(var/obj/machinery/lawrack/R in src.registered_racks)
+			A = get_area(R.loc)
 
-	return laws.Join(glue)
+			laws += "Laws for [R] at [A ? A.name : "...er somewhere?"]:<br>" + R.format_for_logs(glue) +"<br>--------------<br>"
```

Comment:
You can have a null area if you're being deleted (nullspace disposed) if you're seeing this

Reply:
I haven't seen any glitchy stuff with this yet, I just wanted to take precautions in case the rack was constructed in some bizzaro null area. Are you happy leaving it like this, or should I change it?

Reply:
this is resolved by changing it to log_loc anyway, so marking it as such.